rest-api-spec/test/template/30_validate_search_template.yaml

```diff
@@ -0,0 +1,55 @@
+---
+"Indexed Template validate tests":
+
+  - do:
+      put_template:
+        id: "1"
+        body: { "template": { "query": { "match": { "text": "{{my_value}}" } }, "size": "{{my_size}}" } }
+  - match: { _id: "1" }
+
+  - do:
+      indices.refresh: {}
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "foo", "my_size": 20 } }
+
+  - match: { template_output.query.match.text: "foo" }
+  - match: { template_output.size: "20" }
+
+  - do:
+      validate_search_template:
+        body: { "id": "1", "params": { "my_value": "bar", "my_size": 100 } }
+
+  - match: { template_output.query.match.text: "bar" }
+  - match: { template_output.size: "100" }
```

Comment:
Is there a reason the template (`"size": "{{my_size}}"`) and response (`template_output.size: "100"`) are specified as a string, but the input is a numeric (`"my_size": 20`) ?  I don't think this will cause problems for most clients since types are pretty loose, although .Net may be unhappy.  Just curious :)